### DISCUSSION TITLE
Toggle Dark Mode, List Button

### DISCUSSION THREAD

**Discussion by DhammaNotes** [KosmosisDire/obsidian-webpage-export#540] [Ideas]
Greetings,
Test site: https://dhammanotes.github.io/
Currently trying to set up so that dark mode as automatic. After republishing it seems to go to white mode on PC, but dark mode on phone. Is there a means to set the default toggle to dark mode?

Secondly
![2024-11-05_18-11-38](https://github.com/user-attachments/assets/cabef216-96ee-46e4-a4fd-f17feda5438d)
This issue has resolved itself in the beta edition. The title and button for toggle were not rendering correctly.

On the mobile the toggle theme button was rendering on top of the menu button, but have fixed now = (#left-sidebar .topbar-content .... set to end)
![Screenshot_2024-11-05-18-24-26-899_com android chrome](https://github.com/user-attachments/assets/785bbf56-0af1-4533-91f1-ae7908aa50c5)

So the latest beta seems to be working better.

A very helpful plugin, 
With Kindness
Ajahn Kassapa

**Comment by DhammaNotes** [comment]
Greetings,
I am no coder, so this may just require a theme that is black only. But it is very nice to have the css button to change to white.
I tried even setting the theme to dark and disabling the light toggle.

If I access the site from a remote server it loads in light theme.

I would suggest a button to force dark or light theme in the settings pre-export. Even the toggle is set to dark on open, or export without changing theme colour as dark, it still loads theme as light (toggle button does work).

![color BG](https://github.com/user-attachments/assets/c28487f9-c3f6-45ca-994b-d5ce223184f8)

All other bugs seems to of vanished, running very smoothly now as website.

Peace

**Comment by DhammaNotes** [comment]
**So the suggestion is to push a dark or light theme setting as global default.** 
For some reason no even if I use a dark theme only, it still pushes the light theme when I connect with remote sever. I have re-pushed the site so it is free cache.

Great plugin, that is the only error. I suppose I could figure out a custom css/js to push through your plugin but I have not coded in a decade.

![color BG](https://github.com/user-attachments/assets/ce30f7df-df5c-4452-8350-c523eec0cc48)


Peace

**Comment by KosmosisDire** [comment]
Hey, the theme matches the system theme by default of the computer loading the site. So if your computer is set to dark mode it will use dark mode and if light mode it will use light mode. If you change the theme then it should save that theme and use it when the page is reloaded. Is this not the behavior you are seeing?

**Reply by DhammaNotes** [reply]
Great, ok that explains what I am facing then.

The only thing is some themes only have a dark mode css, also some sites are better rendered in one or the other.

So the feature suggestion would be to have a force (dark or light) mode as default on open in the toggle theme settings.

I have been having mixed experience in testing in normal browser and private browser and online browser.

All good wishes

**Reply by DhammaNotes** [reply]
All good. I will try to figure it out now thanks.

I guess something like
@media (prefers-color-scheme: dark)
in snippets

**Reply by DhammaNotes** [reply]
I guess you are correct, rendering a website based on users system preferences is a correct methodology.

Just to note though, that Obsidian has a user preference built in, as many users and themes, have a prefered override.

![Color Checker](https://github.com/user-attachments/assets/63653ff4-b01e-4132-bde9-c42b66c31ecf)

I will work on rebuilding the css, to accommodate this built system preference anyhow.

**Reply by KosmosisDire** [reply]
Yeah, I am aware of the theme picker in obsidian but the site should adapt to the user using the site rather than the author by default. I do agree that there should be a way to force a specific theme by default in the export though!

**Reply by DhammaNotes** [reply]
Greetings again KosmosisDire, 

Any idea what css could be used to force dark theme only? I guess it would use theme-dark tag mentioned in the body class?

`<body class="publish css-settings-manager show-inline-title show-ribbon is-focused is-large-screen theme-dark">`

Would something like `<html data-theme="dark">` work?

Apologies for the noobish question, long time since I have worked with html/css

Great extension.
Peace
DN

**Reply by KosmosisDire** [reply]
You would likely need to write some js to set the class on the body element. I don't have time at the moment to show the full process for this, but I would like to write a general tutorial for writing custom scripts in the future. I will work on making this more easily possible though, thanks
